openmdao/core/system.py

```diff
@@ -808,31 +812,34 @@ def _apply_linear_jac(self, params, unknowns, dparams, dunknowns, dresids, mode)
                              if p not in dparams])
 
         for (unknown, param), J in iteritems(self._jacobian_cache):
+
             if param in states:
                 arg_vec = dunknowns
             else:
                 arg_vec = dparams
 
             # Vectors are flipped during adjoint
-
             try:
                 if isvw:
                     if fwd:
                         vec = dresids._flat(unknown)
-                        vec += J.dot(arg_vec._flat(param))
+                        vec += J.dot(arg_vec[param].flat)
```

Comment:
I originally needed that when I was doing more of the conversion in the vector functions, but it looks like I don't need to do this anymore after my rewrite. I can go back to the _flat calls and still pass all tests.